test/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScannerUnitTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ *  Zed Attack Proxy (ZAP) and its related class files.
+ *
+ *  ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ *  Copyright 2018 The ZAP Development Team
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.zaproxy.zap.extension.wappalyzer;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.model.SiteNode;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscanrulesAlpha.PassiveScannerTest;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Vector;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.given;
+
+public class WappalyzerPassiveScannerUnitTest extends PassiveScannerTest<WappalyzerPassiveScanner> {
+
+	WappalyzerApplicationTestHolder defaultHolder;
+
+	public WappalyzerApplicationTestHolder getDefaultHolder() {
+		if(defaultHolder == null){
+			try {
+				defaultHolder = new WappalyzerApplicationTestHolder();
+				WappalyzerJsonParser parser = new WappalyzerJsonParser();
```

Comment:
We should not use the parser here, if we want to test it we should add a test just for it. We should create/add our own `Application`s (or mock them) instead.